### ISSUE TITLE
Serializing extension types (Request)

### ISSUE THREAD

**Issue by psalmnoelff** [DigDes/SoapCore#824]
So i have some methods that read a request  object, which all have base models. (like from issue #206 )

```
public class Animal
    {
        public Guid RequestId { get; set; }
        public string AnimalCenter { get; set; }
    }
  public class Cat : Animal
    {
        public string Name { get; set; }
    }

    public class Dog : Animal
    {
        public string Name { get; set; }
    }
```

On  my service definition:

```
[ServiceContract]
    public interface ITestService
    {
        [OperationContract]
        ResponseClass1 GetCat(Cat catRequest);

        [OperationContract]
        ResponseClass1 GetDog(Dog dogRequest);

    }
```

However when the wsdl is generated, it  seems to render all properties into a single class, and it does not serialize the extension types (as per issue #209 ).

Is there a flag that we need to enable this? Or should this by default work?

**Comment by psalmnoelff** [comment]
Hi! Just a gentle nudge. I noticed that using DataSerializer, this works by default. Is there a way we can implement this using XMLSerializer?

**Comment by andersjonsson** [comment]
If you look at the changes included in PR #209, support was only added to MetaWCFBodyWriter (the one used with DataContractSerializer).
If you feel like submitting a PR I think you could copy most of those changes from MetaWCFBodyWriter to MetaBodyWriter. Hopefully the structure is similar enough.

Those two classes could probably share much more of the code than they do today...
